src/scripts/main.js

```diff
@@ -1,5 +1,41 @@
 'use strict';
 
+const spider = document.querySelector('.spider');
+const spiderSize = parseFloat(getComputedStyle(spider).height);
+const wall = document.querySelector('.wall');
+const wallSize = parseFloat(getComputedStyle(wall).height);
+
 document.addEventListener('click', e => {
-  // write code here
+  const x = e.offsetX;
+  const y = e.offsetY;
+
+  if (e.target === wall) {
+    let topAdder = 0;
+    let leftAdder = 0;
+    let rightSubtracter = 0;
+    let bottomSubtracter = 0;
+
+    if (x < spiderSize / 2) {
+      leftAdder = spiderSize / 2 - x;
+    }
+
+    if ((x - wallSize + spiderSize / 2) > 0) {
+      rightSubtracter = x - wallSize + spiderSize / 2;
+    }
+
+    if (y < spiderSize / 2) {
+      topAdder = spiderSize / 2 - y;
+    }
+
+    if ((y - wallSize + spiderSize / 2) > 0) {
+      bottomSubtracter = y - wallSize + spiderSize / 2;
+    }
```

Comment:
You repeating yourself two times here. You could create a function that takes adder/substracter  as arguments and return needed numbers. It would reduce the code and you won't repeat yourself